NDTensors/src/tensor/tensor.jl

```diff
@@ -361,16 +361,15 @@ function getdiagindex(T::Tensor{<:Number,N}, ind::Int) where {N}
   return getindex(T, CartesianIndex(ntuple(_ -> ind, Val(N))))
 end
 
+using .DiagonalArrays: diagview
 # TODO: add support for off-diagonals, return
 # block sparse vector instead of dense.
 function diag(tensor::Tensor)
   ## d = NDTensors.similar(T, ElT, (diaglength(T),))
   tensordiag = NDTensors.similar(
     dense(typeof(tensor)), eltype(tensor), (diaglength(tensor),)
   )
-  for n in 1:diaglength(tensor)
-    tensordiag[n] = tensor[n, n]
-  end
+  data(tensordiag) .= diagview(tensor)
```

Comment:
Can this just be:
```suggestion
  tensordiag .= diagview(tensor)
```
? If not I think `array(tensordiag) .= diagview(tensor)` would be better, it is best to avoid `data(...)` whenever there is a reasonable alternative since `data(...)` is meant to be a lower level function.

Reply:
This works, I will keep that in mind for data, thanks!